File: android-app/app/src/main/java/com/example/androidapp/models/FeedColumn.kt
package com.example.androidapp.models

import kotlinx.serialization.Serializable

enum class FeedColumnAlignment {
    LEFT,
    RIGHT,
    CENTER
}

@Serializable
sealed class FeedColumn {

    @Serializable
    data class FeedFavouriteCount(
        val align: FeedColumnAlignment,
        val count: String,
        val signal: Signal?
    ): FeedColumn()

    @Serializable
    data class FeedFavourite(
        val align: FeedColumnAlignment,
        val icon: String,
        val action: FavouriteAction,
        val signal: Signal?
    ): FeedColumn()
}

File: graphql-server/src/components/feed/feed-column.ts
import { feedFavourite } from './feed-favourite';
import { feedFavouriteCount } from './feed-favourite-count';

export const feedColumn = (count, feedId) => {
  const feedFavouriteCache = `feedFavourite-${feedId}`;
  const feedFavouriteCountCache = `feedFavouriteCount-${feedId}`;
  return {
    columns: [
      feedFavourite(count, feedId, feedFavouriteCache),
      feedFavouriteCount(count, feedId, feedFavouriteCountCache)
    ]
  }
}